src/run/revalidate.ts

```diff
@@ -12,7 +13,13 @@ export const nextResponseProxy = (res: ServerResponse, requestContext: RequestCo
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         return async function newRevalidate(...args: any[]) {
           requestContext.didPagesRouterOnDemandRevalidate = true
-          return originalValue?.apply(target, args)
+
+          const result = originalValue?.apply(target, args)
+          if (result && isPromise(result)) {
+            requestContext.trackBackgroundWork(result)
+          }
+
+          return result
```

Comment:
User's CAN await pages router `res.revalidate()`, so if user is doing that, this is not changing anything, but it's quite easy to miss `await` in user code (especially that the promise would resolve to `void` anyway, so user would not be doing anything with `res.revalidate()` return value (other than hopefully `await` it).

This just make sure that if user doesn't `await` it, we will (not blocking a response, just tracking this as background work)